pkg/features/openshift_features.go

```diff
@@ -0,0 +1,26 @@
+package features
+
+import (
+	"k8s.io/apimachinery/pkg/util/runtime"
+	feature "k8s.io/apiserver/pkg/util/feature"
+
+	configv1 "github.com/openshift/api/config/v1"
+	openshiftfeatures "github.com/openshift/library-go/pkg/features"
+)
+
+// OpenShiftFeatureGates contains list of feature gates that will
+// be collectively honored by each of openshift's hyperkube binaries
+var OpenshiftFeatureGates = []configv1.FeatureGateName{
```

Comment:
This file isn't as we discussed in control plane architecture call some time ago. If you look at https://gist.github.com/deads2k/1d58dfa6e556f19ea4c59d0b773bdf86 you'll notice that there are 2 required changes that I don't see here:
1. this file should only have a map of OpenShift feature gates
2. inside `pkg/features/kube_features.go` you should modify this line https://github.com/openshift/kubernetes/blob/master/pkg/features/kube_features.go#L981 to inject k8s and openshift gates.

Reply:
Can you check the implementation of the `InitializeFeatureGates` func that is called towards the end of this file in it's current implementation state. I feel the [that function](https://github.com/openshift/library-go/blob/4109a9e7a43790f6bf68db64171b2257ae20d344/pkg/features/features.go#L88-L121) does it much better and is essentially performing the same map injection that we discussed during the arch call.

Also, the intent here was to ensure nil changes to `pkg/features/kube_features.go` file as that could help keep this patch more maintainable during rebases from upstream. I agree it isn't the same per what we discussed but it's just a syntactical difference in pursuit of helping to keep the Feature list really an array for future features to be directly added without many changes.

Reply:
This is different. We're trying to expand the currently existing kubernetes feature gates mechanism with additional OpenShift specific features. That's the reason we're adding that additional patch file to `pkg/features/` directory.

Reply:
Ok, I think I get it now. :) 
I've updated it accordingly.